# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/powerpc64.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Some CVS changes needed to build dietlibc on powerpc64.

 - pulled by Rene Rebe <rene@exactcode.de>

Only in dietlibc/ppc64/: CVS
diff -u dietlibc-0.29/ppc64/__longjmp.S dietlibc/ppc64/__longjmp.S
--- dietlibc-0.29/ppc64/__longjmp.S	2005-03-23 21:57:53.000000000 +0200
+++ dietlibc/ppc64/__longjmp.S	2005-09-21 10:33:08.000000000 +0300
@@ -11,44 +11,44 @@
 	ld  2,(JB_GPR2*8)(3)
 	ld  0,(JB_LR*8)(3)
 	ld  14,((JB_GPRS+0)*8)(3)
-	lfd 14,((JB_FPRS+0*2)*8)(3)
+	lfd 14,((JB_FPRS+0)*8)(3)
 	ld  15,((JB_GPRS+1)*8)(3)
-	lfd 15,((JB_FPRS+1*2)*8)(3)
+	lfd 15,((JB_FPRS+1)*8)(3)
 	ld  16,((JB_GPRS+2)*8)(3)
-	lfd 16,((JB_FPRS+2*2)*8)(3)
+	lfd 16,((JB_FPRS+2)*8)(3)
 	ld  17,((JB_GPRS+3)*8)(3)
-	lfd 17,((JB_FPRS+3*2)*8)(3)
+	lfd 17,((JB_FPRS+3)*8)(3)
 	ld  18,((JB_GPRS+4)*8)(3)
-	lfd 18,((JB_FPRS+4*2)*8)(3)
+	lfd 18,((JB_FPRS+4)*8)(3)
 	ld  19,((JB_GPRS+5)*8)(3)
-	lfd 19,((JB_FPRS+5*2)*8)(3)
+	lfd 19,((JB_FPRS+5)*8)(3)
 	ld  20,((JB_GPRS+6)*8)(3)
-	lfd 20,((JB_FPRS+6*2)*8)(3)
+	lfd 20,((JB_FPRS+6)*8)(3)
 	mtlr 0
 	ld  21,((JB_GPRS+7)*8)(3)
-	lfd 21,((JB_FPRS+7*2)*8)(3)
+	lfd 21,((JB_FPRS+7)*8)(3)
 	ld  22,((JB_GPRS+8)*8)(3)
-	lfd 22,((JB_FPRS+8*2)*8)(3)
+	lfd 22,((JB_FPRS+8)*8)(3)
 	ld  0,(JB_CR*8)(3)
 	ld  23,((JB_GPRS+9)*8)(3)
-	lfd 23,((JB_FPRS+9*2)*8)(3)
+	lfd 23,((JB_FPRS+9)*8)(3)
 	ld  24,((JB_GPRS+10)*8)(3)
-	lfd 24,((JB_FPRS+10*2)*8)(3)
+	lfd 24,((JB_FPRS+10)*8)(3)
 	ld  25,((JB_GPRS+11)*8)(3)
-	lfd 25,((JB_FPRS+11*2)*8)(3)
+	lfd 25,((JB_FPRS+11)*8)(3)
 	mtcrf 0xFF,0
 	ld  26,((JB_GPRS+12)*8)(3)
-	lfd 26,((JB_FPRS+12*2)*8)(3)
+	lfd 26,((JB_FPRS+12)*8)(3)
 	ld  27,((JB_GPRS+13)*8)(3)
-	lfd 27,((JB_FPRS+13*2)*8)(3)
+	lfd 27,((JB_FPRS+13)*8)(3)
 	ld  28,((JB_GPRS+14)*8)(3)
-	lfd 28,((JB_FPRS+14*2)*8)(3)
+	lfd 28,((JB_FPRS+14)*8)(3)
 	ld  29,((JB_GPRS+15)*8)(3)
-	lfd 29,((JB_FPRS+15*2)*8)(3)
+	lfd 29,((JB_FPRS+15)*8)(3)
 	ld  30,((JB_GPRS+16)*8)(3)
-	lfd 30,((JB_FPRS+16*2)*8)(3)
+	lfd 30,((JB_FPRS+16)*8)(3)
 	ld  31,((JB_GPRS+17)*8)(3)
-	lfd 31,((JB_FPRS+17*2)*8)(3)
+	lfd 31,((JB_FPRS+17)*8)(3)
 	mr 3,4
 	blr
 .size .__longjmp,.-.__longjmp
diff -u dietlibc-0.29/ppc64/setjmp.S dietlibc/ppc64/setjmp.S
--- dietlibc-0.29/ppc64/setjmp.S	2005-03-23 21:57:53.000000000 +0200
+++ dietlibc/ppc64/setjmp.S	2005-09-21 10:33:08.000000000 +0300
@@ -17,43 +17,43 @@
         mflr 0
         std  2,(JB_GPR2*8)(3)
         std  14,((JB_GPRS+0)*8)(3)
-        stfd 14,((JB_FPRS+0*2)*8)(3)
+        stfd 14,((JB_FPRS+0)*8)(3)
         std  0,(JB_LR*8)(3)
         std  15,((JB_GPRS+1)*8)(3)
-        stfd 15,((JB_FPRS+1*2)*8)(3)
+        stfd 15,((JB_FPRS+1)*8)(3)
         mfcr 0
         std  16,((JB_GPRS+2)*8)(3)
-        stfd 16,((JB_FPRS+2*2)*8)(3)
+        stfd 16,((JB_FPRS+2)*8)(3)
         std  0,(JB_CR*8)(3)
         std  17,((JB_GPRS+3)*8)(3)
-        stfd 17,((JB_FPRS+3*2)*8)(3)
+        stfd 17,((JB_FPRS+3)*8)(3)
         std  18,((JB_GPRS+4)*8)(3)
-        stfd 18,((JB_FPRS+4*2)*8)(3)
+        stfd 18,((JB_FPRS+4)*8)(3)
         std  19,((JB_GPRS+5)*8)(3)
-        stfd 19,((JB_FPRS+5*2)*8)(3)
+        stfd 19,((JB_FPRS+5)*8)(3)
         std  20,((JB_GPRS+6)*8)(3)
-        stfd 20,((JB_FPRS+6*2)*8)(3)
+        stfd 20,((JB_FPRS+6)*8)(3)
         std  21,((JB_GPRS+7)*8)(3)
-        stfd 21,((JB_FPRS+7*2)*8)(3)
+        stfd 21,((JB_FPRS+7)*8)(3)
         std  22,((JB_GPRS+8)*8)(3)
-        stfd 22,((JB_FPRS+8*2)*8)(3)
+        stfd 22,((JB_FPRS+8)*8)(3)
         std  23,((JB_GPRS+9)*8)(3)
-        stfd 23,((JB_FPRS+9*2)*8)(3)
+        stfd 23,((JB_FPRS+9)*8)(3)
         std  24,((JB_GPRS+10)*8)(3)
-        stfd 24,((JB_FPRS+10*2)*8)(3)
+        stfd 24,((JB_FPRS+10)*8)(3)
         std  25,((JB_GPRS+11)*8)(3)
-        stfd 25,((JB_FPRS+11*2)*8)(3)
+        stfd 25,((JB_FPRS+11)*8)(3)
         std  26,((JB_GPRS+12)*8)(3)
-        stfd 26,((JB_FPRS+12*2)*8)(3)
+        stfd 26,((JB_FPRS+12)*8)(3)
         std  27,((JB_GPRS+13)*8)(3)
-        stfd 27,((JB_FPRS+13*2)*8)(3)
+        stfd 27,((JB_FPRS+13)*8)(3)
         std  28,((JB_GPRS+14)*8)(3)
-        stfd 28,((JB_FPRS+14*2)*8)(3)
+        stfd 28,((JB_FPRS+14)*8)(3)
         std  29,((JB_GPRS+15)*8)(3)
-        stfd 29,((JB_FPRS+15*2)*8)(3)
+        stfd 29,((JB_FPRS+15)*8)(3)
         std  30,((JB_GPRS+16)*8)(3)
-        stfd 30,((JB_FPRS+16*2)*8)(3)
+        stfd 30,((JB_FPRS+16)*8)(3)
         std  31,((JB_GPRS+17)*8)(3)
-        stfd 31,((JB_FPRS+17*2)*8)(3)
+        stfd 31,((JB_FPRS+17)*8)(3)
         b JUMPTARGET (__sigjmp_save)
 .size .__sigsetjmp,.-.__sigsetjmp
diff -u dietlibc-0.29/ppc64/syscalls.h dietlibc/ppc64/syscalls.h
--- dietlibc-0.29/ppc64/syscalls.h	2005-03-23 21:05:25.000000000 +0200
+++ dietlibc/ppc64/syscalls.h	2005-09-21 10:33:08.000000000 +0300
@@ -270,39 +270,48 @@
 #define __NR_add_key		269
 #define __NR_request_key	270
 #define __NR_keyctl		271
+#define __NR_waitid		272
+#define __NR_ioprio_set		273
+#define __NR_ioprio_get		274
+#define __NR_inotify_init	275
+#define __NR_inotify_add_watch	276
+#define __NR_inotify_rm_watch	277
 
 
-#define diet_proto_weak(wsym) \
-wsym: \
-	.quad	.wsym,.TOC.@tocbase,0;\
-	.previous;\
-	.size	wsym,24;\
-	.type	.wsym,@function;\
-	.globl	.wsym;\
-	.weak	wsym;\
+
+#define __diet_proto_common(sym) \
+	.section ".opd","aw"; \
+	.align	3; \
+sym: \
+	.quad	.sym,.TOC.@tocbase,0; \
+	.previous; \
+	.size	sym,24; \
+	.type	.sym,@function
+
+#define diet_proto_weak(sym) \
+	.weak	sym; \
+	.weak	.sym; \
+	__diet_proto_common(sym)
 
 #define diet_proto(sym) \
-sym:\
-	.quad	.sym,.TOC.@tocbase,0;\
-	.previous;\
-	.size	sym,24;\
-	.type	.sym,@function;\
-	.globl	.sym;\
+	.globl	sym; \
+	.globl	.sym; \
+	__diet_proto_common(sym)
+
 
 #define syscall_weak(name,wsym,sym) \
 .text; \
 diet_proto_weak(wsym); \
 diet_proto(sym); \
-.wsym:\
-	.weak	.wsym;\
+.wsym: \
 .sym: \
 	li	0,__NR_##name; \
-	b __unified_syscall
+	b	__unified_syscall
 
 #define syscall(name,sym) \
 .text; \
 diet_proto(sym); \
 .sym: \
 	li	0,__NR_##name; \
-	b __unified_syscall
+	b	__unified_syscall
 
